File: app/src/main/java/app/we/go/emojidraw/util/EmojiFileReader.java
package app.we.go.emojidraw.util;

import android.content.Context;
import android.support.annotation.NonNull;
import android.util.Log;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.reflect.TypeToken;

import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.Reader;
import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import app.we.go.emojidraw.model.Emoji;

public class EmojiFileReader {

    private List<Emoji> emojiList;

    @Inject
    public EmojiFileReader(Context context) {
        try (InputStream is =context.getAssets().open("emoji-selected.json", Context.MODE_PRIVATE) ){
            Reader reader = new InputStreamReader(is);
            Type listType = new TypeToken<ArrayList<Emoji>>() {}.getType();
            GsonBuilder gsonBuilder = new GsonBuilder();
            Gson gson = gsonBuilder.create();
            emojiList = gson.fromJson(reader, listType);
        } catch (IOException e) {
            Log.e("IO", "Error opening asset with selected emojis");
            emojiList = new ArrayList<>();
        }
    }

    @NonNull
    public List<Emoji> getList() {
        return emojiList;
    }

}

File: app/src/main/java/app/we/go/emojidraw/util/EmojiFileReader.kt
package app.we.go.emojidraw.util

import android.content.Context
import app.we.go.emojidraw.model.Emoji
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.google.gson.reflect.TypeToken
import java.io.InputStreamReader
import java.io.Reader
import javax.inject.Inject

class EmojiFileReader @Inject constructor(context: Context) {

    val emojiList = ArrayList<Emoji>()

    init {
        context.assets.open("emoji-selected.json").use {
            val reader = InputStreamReader(it)
            val gson = GsonBuilder().create()
            emojiList.addAll(gson.fromJson(reader))
        }
    }

    inline fun <reified T> Gson.fromJson(reader: Reader) = this.fromJson<T>(reader, object: TypeToken<T>() {}.type)



}